examples/kalthoff_winkler.cpp

```diff
@@ -18,126 +18,129 @@
+    double rho0 = inputs["density"];
+    double E = inputs["elastic_modulus"];
+    double nu = 1.0 / 3.0;
+    double K = E / ( 3.0 * ( 1.0 - 2.0 * nu ) );
+    double G0 = inputs["fracture_energy"];
+    // double G = E / ( 2.0 * ( 1.0 + nu ) ); // Only for LPS.
+    double delta = inputs["horizon"];
+    delta += 1e-10;
+
+    // ====================================================
+    //                  Discretization
+    // ====================================================
+    // FIXME: set halo width based on delta
+    std::array<double, 3> low_corner = inputs["low_corner"];
+    std::array<double, 3> high_corner = inputs["high_corner"];
+    std::array<int, 3> num_cells = inputs["num_cells"];
+    int m = std::floor( delta /
+                        ( ( high_corner[0] - low_corner[0] ) / num_cells[0] ) );
+    int halo_width = m + 1; // Just to be safe.
+
+    // ====================================================
+    //                   Pre-notches
+    // ====================================================
+    std::array<double, 3> system_size = inputs["system_size"];
+    double height = system_size[0];
+    double width = system_size[1];
+    double thickness = system_size[2];
+    double L_prenotch = height / 2.0;
+    double y_prenotch1 = -width / 8.0;
+    double y_prenotch2 = width / 8.0;
+    double low_x = low_corner[0];
+    double low_z = low_corner[2];
+    Kokkos::Array<double, 3> p01 = { low_x, y_prenotch1, low_z };
+    Kokkos::Array<double, 3> p02 = { low_x, y_prenotch2, low_z };
+    Kokkos::Array<double, 3> v1 = { L_prenotch, 0, 0 };
+    Kokkos::Array<double, 3> v2 = { 0, 0, thickness };
+    Kokkos::Array<Kokkos::Array<double, 3>, 2> notch_positions = { p01, p02 };
+    CabanaPD::Prenotch<2> prenotch( v1, v2, notch_positions );
+
+    // ====================================================
+    //                    Force model
+    // ====================================================
+    using model_type = CabanaPD::ForceModel<CabanaPD::PMB, CabanaPD::Fracture>;
+    model_type force_model( delta, K, G0 );
+    // using model_type =
+    //     CabanaPD::ForceModel<CabanaPD::LPS, CabanaPD::Fracture>;
+    // model_type force_model( delta, K, G, G0 );
+
+    // ====================================================
+    //                 Particle generation
+    // ====================================================
+    // Does not set displacements, velocities, etc.
+    auto particles = std::make_shared<
+        CabanaPD::Particles<memory_space, typename model_type::base_model>>(
+        exec_space(), low_corner, high_corner, num_cells, halo_width );
+
+    // ====================================================
+    //                Boundary conditions
+    // ====================================================
+    double dx = particles->dx[0];
+    double x_bc = -0.5 * height;
+    CabanaPD::RegionBoundary plane(
+        x_bc - dx, x_bc + dx * 1.25, y_prenotch1 - dx * 0.25,
```

Comment:
@streeve Why do we have the factor "1.25" in "dx * 1.25"?

Reply:
Safety factor to avoid floating point comparison issues. But that should be noted in a comment